### PR TITLE
don't use database time

### LINKED ISSUE #2540: Avoid using database time
Due to using a mixture of database and JVM time, some tests sometimes fail when db and JVM times are not in sync. This can happen especially on MacOS, where the DB docker container is running inside a separate virtual machine.

See also: https://github.com/CSCfi/rems/blob/master/test/clj/rems/db/testing.clj#L61

- [x] minimum level: don't use database time when creating data for tests
- [x] never use database time, if it's possible
  - [x] investigate
  - [x] implement

### PR DESCRIPTION
**Pull request by opqdonut**
- drop all now() defaults from schema
- add (time/now) defaults to code where required

fixes #2540

# Checklist for author

Remove items that aren't applicable, check items that are done.

## Reviewability
- [x] Link to issue

## Backwards compatibility
- [x] API is backwards compatible or completely new

## Documentation
- [x] Update changelog if necessary
- [x] ADR for major architectural decisions or experiments
  - should there be one?

## Testing
- [x] Valuable features are integration / browser / acceptance tested automatically
  - test coverage for this stuff seemed pretty good

### CODE REVIEW

**Comment by marharyta** [comment]
I personally have not finished the review of the code, but I believe this is generally a better idea

File: resources/sql/queries.sql [hunk 1]
```diff
@@ -296,9 +296,9 @@ WHERE id = :id;
 
 -- :name create-license-attachment! :insert
 INSERT INTO license_attachment
-(modifierUserId, filename, type, data)
+(modifierUserId, filename, type, data, start)
```
Comment (marharyta) [inline resources/sql/queries.sql:299]: are you planning to add the same code to ```organisations``` ?

Comment (opqdonut) [reply resources/sql/queries.sql:299]: It's there already! It turns out the default values in audit_log and organization were not used: REMS always specified the value explicitly.

Comment (marharyta) [reply resources/sql/queries.sql:299]: ok, so we did have this approach already? Good to refactor then!

Comment (opqdonut) [reply resources/sql/queries.sql:299]: Yeah it was kind of a belt-and-suspenders thing where most timestamps were created in REMS, but some were left up to the db defaults.

File: CHANGELOG.md [hunk 2]
```diff
@@ -11,6 +11,7 @@ Changes since v2.15
 ### Changes
 - REMS no longer sends the Server: HTTP header to avoid leaking version information. (#2216)
 - Text descriptions of some events in the log were phrased better. The created event also shows the original external id. (#2614)
+- REMS no longer uses the database time when creating rows. Requires migration, but does not change visible behaviour. (#2540)
```
Comment (Macroz) [inline CHANGELOG.md:14]: Maybe would mention that it uses application server time now consistently instead.

**Review by Macroz — APPROVED** [review]

File: resources/sql/queries.sql (hunk 1 above)
Comment (Macroz) [reply resources/sql/queries.sql:299]: The db defaults make sense for helping with the SQL console use. I think they can be left out. The middle ground is to have the defaults and always make sure you don't use the default in the app.

**Comment by Macroz** [comment]
Looks good! Maybe a quick couple line ADR about the database design where we would only have this point for now?

**Comment by marharyta** [comment]
Looks good to me too.

**Review by marharyta — APPROVED** [review]

**Review by Macroz — APPROVED** [review]

**Pull request merged**